spec.yaml

```
@@ -575,158 +572,156 @@ components:
+                    be accessible by appending their relative path to that URI.
+                viewMode:
+                  type: string
+                  description: |
+                    The permissions granted to the sharee.
+                    - `view` allows access to the web app in view-only mode.
+                    - `read` allows read and download access via the web app.
+                    - `write` allows full editing rights via the web app.
+                  enum:
+                    - view
+                    - read
+                    - write
+                sharedSecret:
+                  type: string
+                  description: >
+                    An optional secret to be used to access the remote web app, such as
+                    a bearer token. To prevent leaking it in logs it MUST NOT appear
+                    in any URI. If a `code` is provided, then the sending host MUST
+                    accept the short-lived bearer token when serving the web app,
+                    which can be exchanged in the code flow interaction. The exchange
+                    MAY already have happened if the recipient accessed the underlying
+                    resource via WebDAV, in a multi-protocol scenario. In this case,
+                    the `sharedSecret` SHOULD be omitted.
+            datatx:
+              type: object
+              properties:
+                sharedSecret:
+                  type: string
+                  description: >
+                    An optional secret to be used to access the resource, such as
+                    a bearer token. If a `code` is provided, it SHOULD be used
+                    instead via the code flow interaction, and the `sharedSecret`
+                    SHOULD be omitted. To prevent leaking it in logs it MUST NOT
+                    appear in any URI.
+                srcUri:
+                  type: string
+                  description: >
+                    An URI to access the resource at the sending server. The URI
+                    SHOULD be relative, such as a key or a UUID, in which case the
+                    prefix exposed by the `/.well-known/ocm` endpoint SHOULD be used
+                    to access the resource, or it MAY be absolute, including a
+                    hostname. Similar considerations as for the `webdav` case apply.
+                size:
+                  type: integer
+                  description: >
+                    The size of the file to be transferred from the sending
+                    server.
+
+          additionalProperties:
+            type: object
+            description: >
+              Any optional additional protocols supported for this resource
+              MAY
+              be provided here, along with their custom payload. Appropriate
+              capabilities MUST be advertised in order for a sender to ensure
```

Bringing back this from #181:
```suggestion
              capabilities SHOULD be advertised in order for a sender to ensure
```